scripts/build-local.py

```diff
@@ -0,0 +1,165 @@
+                    '--directory',
+                    help='Path to kernel source',
+                    required=True)
+parser.add_argument('-d',
+                    '--debug',
+                    action='store_true',
+                    help='Show debugging messages')
+parser.add_argument('-f',
+                    '--files',
+                    help='TuxSuite YAML files to build',
+                    nargs='+',
+                    required=True)
+parser.add_argument('-j',
+                    '--jobs',
+                    help='Jobs to build (default: build all jobs)',
+                    nargs='+')
+parser.add_argument(
+    '-o',
+    '--output-dir',
+    default=default_output_dir,
+    help=f"Output folder for TuxMake files (default: {default_output_dir})")
+parser.add_argument('-v',
+                    '--verbose',
+                    action='store_true',
+                    help="Show tuxmake's output")
+args = parser.parse_args()
+
+if not (tree := Path(args.directory).resolve()).exists():
+    raise FileNotFoundError(
+        f"Requested tree ('{tree}'), derived from provided directory ('{args.directory}'), could not be found?"
+    )
+
+if shutil.which('podman'):
+    runtime = 'podman'
+elif shutil.which('docker'):
+    runtime = 'docker'
+else:
+    raise RuntimeError(
+        "tuxmake requires either podman or docker for versioned toolchains but neither could be found on your system!"
+    )
+
+# Combine all jobs into one object for easy iteration
+jobs = {}
+for file_str in args.files:
+    with open(file_str, encoding='utf-8') as file:
+        config = yaml.safe_load(file)
+
+    current_jobs = config['jobs']
+
+    for job in config['jobs']:
+        name = job['name']
+        if args.jobs and name not in args.jobs:
+            # This can be noisy under normal circumstances so only show it if
+            # debugging is enabled.
+            if args.debug:
+                jobs_join = "', '"
+                print(
+                    f"D: {Path(file_str).name}: Job '{name}' was not in requested jobs ('{jobs_join.join(args.jobs)}'), skipping..."
+                )
+            continue
+
+        builds = job['builds']
+        if name in jobs:
+            jobs[name] += builds
+        else:
+            jobs[name] = copy.deepcopy(builds)
+
+build_dir = Path(args.build_dir).resolve()
+output_dir = Path(args.output_dir).resolve()
+for name, builds in jobs.items():
+    print(f"I: Executing job '{name}' in {tree}")
+    for build in builds:
+        kconfig = build['kconfig']
+        target_arch = build['target_arch']
+        toolchain = build['toolchain']
+
+        cfg_str = '+'.join(kconfig) if isinstance(kconfig, list) else kconfig
+        print(f"I: Building {target_arch} {cfg_str} ({toolchain})... ",
+              end='\n' if args.verbose else '',
+              flush=True)
+
+        if build_dir.exists():
+            shutil.rmtree(build_dir)
+        build_dir.mkdir(exist_ok=True, parents=True)
```

Comment:
probably don't need `exist_ok=True` anymore? If it existed, it would have been removed on the line above.  Though perhaps TOCTOU is a concern between the two statements?

Reply:
Yeah, this was paranoia over `shutil.rmtree()` failing but I did not specify `ignore_errors=True` for that call so if it fails, an exception will be raised. I think I will just remove the `exist_ok=True` from the `mkdir()` call. Alternatively, I could do:

```
shutil.rmtree(build_dir, ignore_errors=True)
build_dir.mkdir(exist_ok=True, parents=True)
```

but I think removing `exist_ok=True` is a bit better. Done: e3a1493447355709daeb9bae4f34c9987a40446b